### DISCUSSION TITLE
Can you clarify how does modifying an original file (through exif-push) affects iCloud synching?

### DISCUSSION THREAD

**Discussion by santiagoGPNC** [RhetTbull/osxphotos#1260] [Q&A] [answered]
I have a photo library that has iCloud sync enabled. For a long time I added the location metadata to photos taken with my DSLR without knowing that does not affect the actual file (which I want to backup into a NAS now).
But Im wondering how the photos.app react to this? I don't think the app expects changes in the original file, so Im not sure if any change to the exif files will be synced to the cloud and other devices.

I gave it a try and at least it didn't seem to happen straight away, but iCloud is a black box on how it decides when to synchronize photos so I though I should ask.

Will the photos app update to iCloud the files I update using push-exif? Is there something I can do to force it? I would prefer not having to remove and readd them since it's a lot of them.

Thanks for your help!

**Comment by RhetTbull** [comment, marked as answer]
I cannot know with certainty because as you said, iCloud is a black box, but I do understand Photos well and I am fairly certain that modifying the local copy of an image does NOT affect the copy on iCloud. Thus if Photos ever syncs an image (for example, it was changed on another device) it's likely the exif changes get overwritten. push-exif is really only intended for non-iCloud libraries and will warn you if you run it on an iCloud library. I think it's "likely safe" but I wouldn't do it on my own library. 

> without knowing that does not affect the actual file (which I want to backup into a NAS now).

`osxphotos export` can handle this for you with the `--exiftool` option which does what push-exif does but on the exported copy of a file. It's not quite as configurable as `push-exif` (it's all or nothing) but it does preserve the maximum amount of metadata. The other options like `--keyword-template` and `--person-keyword` allow you to customize it.

**Reply by santiagoGPNC** [reply]
thanks. I was afraid that were the case. I tried using push-exif on a photo and then edit it a bit. What I saw is that the edited version got pushed, but exporting the original in other device did not reflect the new metadata. As you said, it seems once the original is in iCloud, it does not expect any changes in the original file (which make sense a bit since they have the option to revert changes or export the original without modifications).

I was hoping I could add the metadata to those files and backthem up, but seems like I would probably need to remove them and add them back to my icloud library.

Thank you for the clarification.
